js/ui/paygUnlockDialog.js

```diff
@@ -328,6 +333,53 @@ var PaygUnlockDialog = new Lang.Class({
         }
     },
 
+    _onSuccess: function(mainBox) {
+            mainBox.remove_all_children();
+
+            //Measuring the ceiling of the timeRemainingSecs because by the time this runs, some time has already elapsed
+            let timeString = Main.paygManager._convertTimeToString(Math.ceil(Main.paygManager._timeRemainingSecs()));
```

Comment:
This is kind of an interesting problem because if the system is under heavy load, it could conceivably take more than 1 second for us to get to this point, at which point `Math.ceil` will not save us and the user will be upset because they paid for 1 day and it shows them 23 hours 59 minutes or similar.